src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a TaskList object. The TaskList is able to add tasks, delete tasks, mark tasks as done,
+ * as well as print out all the tasks in its list.
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+    private int counter = 0;
```

Comment:
Since this counter is not directly accessible outside and is not be used for any check you could maybe remove it.